### ISSUE TITLE
Compiling with third-party typing libraries may not display errors in VSCode

### ISSUE THREAD

**Issue by ilsubyeega** [nvarner/typst-lsp#376]
- **Component**: 
    - VSCode Extension
- **Extension version**: v0.11.0
- **OS version and name**: Windows 11 (x64)

- I am on the [latest](https://github.com/nvarner/typst-lsp/tags) stable version of the extension/LSP.
- I have searched the [issues](https://github.com/nvarner/typst-lsp/issues) of this repo and believe that this is not a duplicate.

## Issue
I am using cetz v0.1.2, available at [GitHub](https://github.com/johannes-wolf/cetz) in Visual Studio Code, and I noticed that no error messages are displayed in the editor when I have invalid code. Below is the code that is considered valid:

```typst
#import "@preview/cetz:0.1.2": canvas, draw, tree

#canvas(length: 1cm, {
  import draw: *
  circle((0,0), radius: .5)
  content((0,0), "test")
})
```

However, when I remove the `(0,0),` from the `content` function, like this:

```typst
#import "@preview/cetz:0.1.2": canvas, draw, tree

#canvas(length: 1cm, {
  import draw: *
  circle((0,0), radius: .5)
  content("test") // <----
})
```

...I expect to see a compilation error because the coordinates are required for the `content` function. While the code does fail to compile as expected, there is no error message displayed in the output tab or the editor view within VSCode.

However, if I compile the code manually using the command line interface (CLI) with typst version 0.9.0, the error is displayed. 

Here’s an image demonstrating the error message when using CLI compilation: ![Error Image on GitHub](https://github.com/nvarner/typst-lsp/assets/37479424/316adee2-6f83-4d04-889e-46d396f45125).

To summarize, it seems that the only way to detect errors in the code with cetz in VSCode is to compile manually using the CLI. This is not an ideal workflow and should probably be addressed.

Thank you.
### Logs
- From VSCode output tab:
```
[Info  - 1:12:48 PM] src\server\export.rs:14 { message = exporting PDF; pdf_uri = file:///c%3A/Users/ilsub/Projects/my-typst/lsp-bug.pdf; }
[Info  - 1:12:48 PM] src\server\export.rs:27 { message = PDF export complete; }
[Error - 1:15:25 PM] src\server\lsp.rs:298 { message = could not handle source save; err = failed to generate document after compilation; uri = file:///c%3A/Users/ilsub/Projects/my-typst/lsp-bug.typ; }
```
